keps/sig-storage/20191125-bucket-provisioning.md

```diff
@@ -364,7 +364,7 @@ metadata:
 
 1. `name`: the BA name is generated by COSI in this format: _"<BAR.namespace*>-<BAR.name*>-<uuid>"_. The uuid is unique within a cluster. `*` the first 10 characters of the namespace and the name are used.
 1. `labels`: added COSI.  Key’s value should be the provisioner name. Characters that do not adhere to [Kubernetes label conventions](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set) will be converted to ‘-’.
-1. `finalizers`: added by COSI to defer `BucketAccess` deletion until backend deletion ops succeed.
+1. `finalizers`: added by COSI to defer `BucketAccess` deletion until the pod that uses this access via `BucketAccessRequest` is deleted.
```

Comment:
Question: if the BAR still exists and the pod terminates should cosi delete the BA? If yes, then we have an orphaned BAR. If no, then at least the BAR and BA are consistent. What do you think?

Reply:
I think normal deletion process is user deletes the BAR and BA should be deleted. If Pod is delete it will not delete the BA. only BA is waiting on the Pod.

Reply:
So, "_...to defer `BucketAccess` deletion until the pod that uses this access via `BucketAccessRequest` is deleted_" should be changed to:

"_...to defer `BucketAccess` deletion until the pod that uses this access terminates, and the related `BucketAccessRequest` is deleted._" in order to reflect that the BAR also needs to be deleted. Do you think this correct?